databricks/koalas/base.py

```diff
@@ -118,22 +119,27 @@ def wrapper(self, *args):
     return wrapper
 
 
-class IndexOpsMixin(object):
+class IndexOpsMixin(object, metaclass=ABCMeta):
     """common ops mixin to support a unified interface / docs for Series / Index
 
     Assuming there are following attributes or properties and function.
 
-    :ivar _kdf: Parent's Koalas DataFrame
-    :type _kdf: ks.DataFrame
-    :ivar spark: Spark-related features
-    :type spark: SparkIndexOpsMethods
+    :ivar _anchor: Parent's Koalas DataFrame
+    :type _anchor: ks.DataFrame
     """
 
-    def __init__(self, internal: InternalFrame, kdf):
-        assert internal is not None
-        assert kdf is not None and isinstance(kdf, DataFrame)
-        self._internal = internal  # type: InternalFrame
-        self._kdf = kdf
+    def __init__(self, anchor: DataFrame):
+        assert anchor is not None
+        self._anchor = anchor
+
+    @property
+    @abstractmethod
+    def _internal(self) -> InternalFrame:
+        pass
+
+    @property
+    def _kdf(self) -> DataFrame:
```

Comment:
I have some question here !

For `IndexOpsMixin` class, we have `_anchor` and `_kdf` to indicates DataFrame that corresponds with `Series` or `Index`, and they are exactly same.

Is there a special reason that we have both property even they indicates same object ?